src/App.tsx

```diff
@@ -1,18 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+type Item = {
+  id: number,
+  title: string,
+};
+
+const items = getNumbers(1, 42).map(n => `Item ${n}`);
+
+const preparedItems: Item[] = items
+  .map((item, index) => ({ id: index, title: item }));
+
+function getVisibleItems(allItems: Item[], start: number, end: number) {
+  return allItems.slice(start, end + 1);
+}
 
 export const App: React.FC = () => {
+  const [itemsOnPage, setitemsOnPage] = useState(5);
+  const [selectedPage, setSelectedPage] = useState(1);
+
+  const start = (selectedPage - 1) * itemsOnPage;
+  const end = Math.min(start + itemsOnPage - 1, preparedItems.length - 1);
+
+  const visibleItems = getVisibleItems(preparedItems, start, end);
+
+  const handleOnPageChange = (pageNumber: number) => {
+    if (selectedPage !== pageNumber) {
+      setSelectedPage(pageNumber);
```

Comment:
If you want to update the state according to the previous value, do it according to these rules:
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state